src/btree/bt_curnext.c

```diff
@@ -479,7 +480,8 @@ __cursor_row_next(
                 WT_STAT_CONN_DSRC_INCR(session, cursor_bounds_next_early_exit);
             WT_RET(ret);
 
-            WT_RET(__wt_txn_read_upd_list(session, cbt, key, WT_RECNO_OOB, ins->upd));
+            upd = (WT_UPDATE *)__wt_tsan_suppress_load_pointer((void **)&ins->upd);
```

Comment:
This here feels error prone, is there any other way we can format this? Or create a macro for this.

Reply:
I can create separate functions for every case, e.g. for this case:

```suggestion
            upd = __wt_tsan_suppress_load_wt_update_pointer(&ins->upd);
```

Would it work for you?

Reply:
The existing `atomic_load_pointer` implementation doesn't require typing for every load and avoids the (void **) casting. Can we utilize that method to implement the TSAN load pointer equivalent?

Reply:
Unfortunately, no, we cannot, since both `__wt_atomic_load_pointer` and `__wt_atomic_store_pointer` are macros. That’s why they don’t require type casting, and this also prevents them from participating in suppression (as they don’t exist after preprocessing).

Reply:
In that case I'd create a function for every case, I think `_Generic` would be the actual solution but MSVC support is problematic.

Reply:
I added a separate function for every case, please take a look